### ISSUE TITLE
yapet: ftbfs with GCC-10

### ISSUE THREAD

**Issue by carnil** [RafaelOstertag/yapet#20]
We have the following bug reported to the Debian package of yapet,
c.f. https://bugs.debian.org/958008

It doesn't seem to be a bug in the packaging, so you may want to take
a look. Thanks!

------8<-----------8<-----------8<-----------8<-----------8<-----

```
Package: src:yapet
Version: 2.3-1
Severity: normal
Tags: sid bullseye
User: debian-gcc@lists.debian.org
Usertags: ftbfs-gcc-10

Please keep this issue open in the bug tracker for the package it
was filed for.  If a fix in another package is required, please
file a bug for the other package (or clone), and add a block in this
package. Please keep the issue open until the package can be built in
a follow-up test rebuild.

The package fails to build in a test rebuild on at least amd64 with
gcc-10/g++-10, but succeeds to build with gcc-9/g++-9. The
severity of this report will be raised before the bullseye release,
so nothing has to be done for the buster release.

The full build log can be found at:
http://people.debian.org/~doko/logs/gcc10-20200225/yapet_2.3-1_unstable_gcc10.log
The last lines of the build log are at the end of this report.

To build with GCC 10, either set CC=gcc-10 CXX=g++-10 explicitly,
or install the gcc, g++, gfortran, ... packages from experimental.

  apt-get -t=experimental install g++ 

Common build failures are new warnings resulting in build failures with
-Werror turned on, or new/dropped symbols in Debian symbols files.
For other C/C++ related build failures see the porting guide at
http://gcc.gnu.org/gcc-10/porting_to.html

[...]
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/pwgen -I../../../src/libs/cfg -I../../../src/libs/globals -I../../../src/libs/consts -I../../../src/libs/metadata -I../../../src/libs/exceptions -I../../../src/libs/file -I../../../src/libs/passwordrecord -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../libyacurs/include -I../../../libyacurs/include -I../../.. -I../../../src/libs -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c cryptofactoryhelper.cc  -fPIC -DPIC -o .libs/cryptofactoryhelper.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/pwgen -I../../../src/libs/cfg -I../../../src/libs/globals -I../../../src/libs/consts -I../../../src/libs/metadata -I../../../src/libs/exceptions -I../../../src/libs/file -I../../../src/libs/passwordrecord -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../libyacurs/include -I../../../libyacurs/include -I../../.. -I../../../src/libs -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c blowfishfactory.cc -o blowfishfactory.o >/dev/null 2>&1
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/pwgen -I../../../src/libs/cfg -I../../../src/libs/globals -I../../../src/libs/consts -I../../../src/libs/metadata -I../../../src/libs/exceptions -I../../../src/libs/file -I../../../src/libs/passwordrecord -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../libyacurs/include -I../../../libyacurs/include -I../../.. -I../../../src/libs -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c aes256factory.cc -o aes256factory.o >/dev/null 2>&1
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/pwgen -I../../../src/libs/cfg -I../../../src/libs/globals -I../../../src/libs/consts -I../../../src/libs/metadata -I../../../src/libs/exceptions -I../../../src/libs/file -I../../../src/libs/passwordrecord -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../libyacurs/include -I../../../libyacurs/include -I../../.. -I../../../src/libs -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c cryptofactoryhelper.cc -o cryptofactoryhelper.o >/dev/null 2>&1
/bin/bash ../../../libtool  --tag=CXX   --mode=link g++  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security  -lssl -lcrypto -largon2 -lrt -ldl -Wl,-z,relro -Wl,-z,now -o libyapet-crypt.la  key448.lo key256.lo file.lo blowfish.lo aes256.lo crypto.lo blowfishfactory.lo aes256factory.lo cryptofactoryhelper.lo  
libtool: link: ar cr .libs/libyapet-crypt.a .libs/key448.o .libs/key256.o .libs/file.o .libs/blowfish.o .libs/aes256.o .libs/crypto.o .libs/blowfishfactory.o .libs/aes256factory.o .libs/cryptofactoryhelper.o 
libtool: link: ranlib .libs/libyapet-crypt.a
libtool: link: ( cd ".libs" && rm -f "libyapet-crypt.la" && ln -s "../libyapet-crypt.la" "libyapet-crypt.la" )
make[5]: Leaving directory '/<<PKGBUILDDIR>>/src/libs/crypt'
Making all in glue
make[5]: Entering directory '/<<PKGBUILDDIR>>/src/libs/glue'
/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c -o loadfile.lo loadfile.cc
/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c -o promptpassword.lo promptpassword.cc
/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c -o createfile.lo createfile.cc
/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c -o changepassword.lo changepassword.cc
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c loadfile.cc  -fPIC -DPIC -o .libs/loadfile.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c promptpassword.cc  -fPIC -DPIC -o .libs/promptpassword.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c createfile.cc  -fPIC -DPIC -o .libs/createfile.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c changepassword.cc  -fPIC -DPIC -o .libs/changepassword.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c promptpassword.cc -o promptpassword.o >/dev/null 2>&1
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c loadfile.cc -o loadfile.o >/dev/null 2>&1
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c createfile.cc -o createfile.o >/dev/null 2>&1
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c changepassword.cc -o changepassword.o >/dev/null 2>&1
/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c -o opencmdlinefile.lo opencmdlinefile.cc
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c opencmdlinefile.cc  -fPIC -DPIC -o .libs/opencmdlinefile.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/globals -I../../../src/libs/exceptions -I../../../src/libs/metadata -I../../../src/libs/crypt -I../../../src/libs/pwgen -I../../../src/libs/ui -I../../../src/libs/consts -I../../../src/libs/cfg -I../../../src/libs/file -I../../../src/libs/utils -I../../../src/libs/interfaces -I../../../src/libs/passwordrecord -I../../../libyacurs/include -I../../../libyacurs/include -I../../../src/libs -I../../.. -DLOCALEDIR=\"/usr/share/locale\" -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c opencmdlinefile.cc -o opencmdlinefile.o >/dev/null 2>&1
/bin/bash ../../../libtool  --tag=CXX   --mode=link g++  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security  -Wl,-z,relro -Wl,-z,now -o libyapet-glue.la  loadfile.lo promptpassword.lo createfile.lo changepassword.lo opencmdlinefile.lo  
libtool: link: ar cr .libs/libyapet-glue.a .libs/loadfile.o .libs/promptpassword.o .libs/createfile.o .libs/changepassword.o .libs/opencmdlinefile.o 
libtool: link: ranlib .libs/libyapet-glue.a
libtool: link: ( cd ".libs" && rm -f "libyapet-glue.la" && ln -s "../libyapet-glue.la" "libyapet-glue.la" )
make[5]: Leaving directory '/<<PKGBUILDDIR>>/src/libs/glue'
Making all in pwgen
make[5]: Entering directory '/<<PKGBUILDDIR>>/src/libs/pwgen'
/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../../src/libs/utils -I../../../src/libs/consts -DLOCALEDIR=\"/usr/share/locale\" -I../../.. -DNDEBUG  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c -o rngengine.lo rngengine.cc
/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../../src/libs/utils -I../../../src/libs/consts -DLOCALEDIR=\"/usr/share/locale\" -I../../.. -DNDEBUG  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c -o rng.lo rng.cc
/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../../src/libs/utils -I../../../src/libs/consts -DLOCALEDIR=\"/usr/share/locale\" -I../../.. -DNDEBUG  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c -o pwgen.lo pwgen.cc
/bin/bash ../../../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I. -I../../..  -I../../../src/libs/utils -I../../../src/libs/consts -DLOCALEDIR=\"/usr/share/locale\" -I../../.. -DNDEBUG  -Wdate-time -D_FORTIFY_SOURCE=2  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c -o characterpool.lo characterpool.cc
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/utils -I../../../src/libs/consts -DLOCALEDIR=\"/usr/share/locale\" -I../../.. -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c rngengine.cc  -fPIC -DPIC -o .libs/rngengine.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/utils -I../../../src/libs/consts -DLOCALEDIR=\"/usr/share/locale\" -I../../.. -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c pwgen.cc  -fPIC -DPIC -o .libs/pwgen.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/utils -I../../../src/libs/consts -DLOCALEDIR=\"/usr/share/locale\" -I../../.. -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c rng.cc  -fPIC -DPIC -o .libs/rng.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/utils -I../../../src/libs/consts -DLOCALEDIR=\"/usr/share/locale\" -I../../.. -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c characterpool.cc  -fPIC -DPIC -o .libs/characterpool.o
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/utils -I../../../src/libs/consts -DLOCALEDIR=\"/usr/share/locale\" -I../../.. -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c characterpool.cc -o characterpool.o >/dev/null 2>&1
rngengine.cc: In function âint {anonymous}::openDevUrandom()â:
rngengine.cc:59:20: error: âruntime_errorâ is not a member of âstdâ
   59 |         throw std::runtime_error(msg);
      |                    ^~~~~~~~~~~~~
rngengine.cc: In member function âvoid yapet::pwgen::RngEngine::fillCache()â:
rngengine.cc:74:20: error: âruntime_errorâ is not a member of âstdâ
   74 |         throw std::runtime_error(_("Error reading random bytes from file"));
      |                    ^~~~~~~~~~~~~
make[5]: *** [Makefile:451: rngengine.lo] Error 1
make[5]: *** Waiting for unfinished jobs....
pwgen.cc: In member function âyapet::SecureArray yapet::pwgen::PasswordGenerator::generatePassword(int)â:
pwgen.cc:143:20: error: âinvalid_argumentâ is not a member of âstdâ
  143 |         throw std::invalid_argument(msg);
      |                    ^~~~~~~~~~~~~~~~
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../.. -I../../../src/libs/utils -I../../../src/libs/consts -DLOCALEDIR=\"/usr/share/locale\" -I../../.. -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -c rng.cc -o rng.o >/dev/null 2>&1
make[5]: *** [Makefile:451: pwgen.lo] Error 1
make[5]: Leaving directory '/<<PKGBUILDDIR>>/src/libs/pwgen'
make[4]: *** [Makefile:500: all-recursive] Error 1
make[4]: Leaving directory '/<<PKGBUILDDIR>>/src/libs'
make[3]: *** [Makefile:389: all-recursive] Error 1
make[3]: Leaving directory '/<<PKGBUILDDIR>>/src'
make[2]: *** [Makefile:538: all-recursive] Error 1
make[2]: Leaving directory '/<<PKGBUILDDIR>>'
make[1]: *** [Makefile:428: all] Error 2
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
dh_auto_build: error: make -j4 returned exit code 2
make: *** [debian/rules:7: binary] Error 25
dpkg-buildpackage: error: debian/rules binary subprocess returned exit status 2


```
------8<-----------8<-----------8<-----------8<-----------8<-----

Fedora has already patched it with
https://src.fedoraproject.org/rpms/yapet/blob/b33bd774672f2c931a654c92f3ddc3eda99f2f69/f/gcc10.patch

Thanks for considering,
  Salvatore Bonaccorso,

**Comment by RafaelOstertag** [comment]
Thank you for bringing this to my attention. It has been fixed with version 2.4

**Comment by carnil** [comment]
@RafaelOstertag thanks!
